EIPS/eip-1775.md

```diff
@@ -0,0 +1,571 @@
+
+## Motivation
+<!--The motivation is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. EIP submissions without sufficient motivation may be rejected outright.-->
+Wallets developers have agreed on an HD derivation path for ethereum accounts using BIP32, BIP44, SLIP44, [ERC84](https://github.com/ethereum/EIPs/issues/84). Web3 wallets have implemented in a roughly similar way the rpc eth api. EIP 1102 introduced privacy through non automatic opt-in of a wallet account into an app increasing privacy.
+
+However several limitations remain in order to allow for proper design and UX for crypto permissioned apps.
+
+Most of GUI based current wallets don't allow to:
+* being able to automatically and effortlessly use different keys / accounts for each apps,
+* being able to sign some app's action without prompting the user with the same level of security as sending funds from their main accounts,
+* being able to use throwable keys to improve anonymity,
+* effortlessly signing transactions for an app without broadcasting these while still being able to perform other transaction signing as usual from their main accounts,
+* All this while being fully restorable using the user's mnemonic or hardware wallet and the HD Path determined uniquely by the app's ens name.
+
+We try to overcome these limitations by introducing a new account's type, app keys, made to be used along side the existing main accounts.
+
+These new app keys can permit to give more power and flexibility to the crypto apps developers. This can allow to improve a lot the UX of crypto dapps and to create new designs that were not possible before leveraging the ability to create and handle many accounts, to presign messages and broadcast them later. These features were not compatible with the level of security we were requesting for main accounts that hold most of an user's funds.
+
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+
+### Applications
+
+An app is a website (or other) that would like to request from a wallet to access app keys. It can be any form of cryptography/identity relying application, ethereum but not only.
+
+Once connected to a wallet, an application can request to access a set of accounts derived exclusively for that application using the hierarchical deterministic (HD) paths.
+
+### Applications' HD path
+
+Using the BIP 32 and BIP 43 standards, we propose to use the following HD path for each app keys:
+
+`m / [EIP Number]' / [persona path]' / [application uniquely assigned path]' / [app's custom subpath]`
```

Comment:
Doing this will collide with Bitcoin and any number of other coins. See [this Bitcoin PR](https://github.com/bitcoin/bips/pull/523) for a suggestion of how to handle things; following that scheme it would be `m/43'/60'/[EIP Number]'/...`.

Reply:
Agreed that we should avoind purpose field collision, however I think there should be a list of BIP 43' purpose codes.

I don't think it would be a solution to make this a child of 43'/60' since this is not Ethereum specific. This is a cross-crypto standard (even not only cryptocurrency related).

What about suggesting the creation of a list of BIP43 purpose codes and to suggest developpers to refer to it before creating new purpose code. It would be similar to BIP44 and SLIP44.

In fact that's already what is happening by usage, several other purpose codes have been used:

see here for a temptative list:
https://github.com/ethereum/EIPs/blob/1ae40193625ddaf9f1ac1c28d661fbce5ed5feac/EIPS/eip-1775.md#hd-derivation-path-purpose-field

and here:
https://bitcoin.stackexchange.com/questions/60470/is-there-a-comprehensive-list-of-registered-bip43-purposes

In any case, what matters to me here is that we don't make these paths rely on a coin_type since the keys derived for applications or the cross applications keys (https://ethereum-magicians.org/t/eip-erc-app-keys-application-specific-wallet-accounts/2742/25?u=bunjin) are not dependent on a single cryptocurrency (and may even be cryptocurrency less).

Reply:
Resolved by using what you suggested for now and mentioning my proposals in the alternative/rationale part.